docker/test.py

```diff
@@ -0,0 +1,32 @@
+import requests
+import os
+
+zks = os.getenv('ZOO_SERVERS')
+zk_stats = {}
+
+for zk in zks.split(' '):
+    zk_host = zk.split('=')[1].split(':')[0]
+    zk_stats[zk_host] = requests.get('http://{}:{}/commands/monitor'.format(zk_host, 8080), timeout=0.01).json()
+
+leader = []
+followers = []
+synced_followers = 0
+
+for zk in zk_stats:
+    stats = zk_stats[zk]
+    if 'server_state' in stats:
+        if stats['server_state'] == 'leader':
+            leader.append(zk)
+            synced_followers = stats['synced_followers']
+        else:
+            followers.append(zk)
+
+if synced_followers < 2 and (len(leader) > 1 or len(leader) < 1 or len(followers) < 2 or len(followers) > 2):
```

Comment:
I think you meant `or`

Reply:
No, it also checks the follower nodes and it doesn't have the synced_followers value so it will always fail.
`
root@ip-10-11-102-237:/apache-zookeeper-3.5.6-bin# echo mntr | nc localhost 2181
zk_version      3.5.6-c11b7e26bc554b8523dc929761dd28808913f091, built on 10/08/2019 20:18 GMT
zk_avg_latency  0
zk_max_latency  0
zk_min_latency  0
zk_packets_received     1
zk_packets_sent 0
zk_num_alive_connections        1
zk_outstanding_requests 0
zk_server_state follower
zk_znode_count  5
zk_watch_count  0
zk_ephemerals_count     0
zk_approximate_data_size        281
zk_open_file_descriptor_count   57
zk_max_file_descriptor_count    65536
root@ip-10-11-102-237:/apache-zookeeper-3.5.6-bin# echo stat | nc localhost 2181 | grep Mode
Mode: follower
`

Reply:
I just checked, Yes you are correct. That check happens after the loop.